main.cpp

```diff
@@ -259,72 +250,104 @@ struct HeapA
     }
 };
 
-#include <iostream>
 struct IntType;
 struct DoubleType;
 
 struct FloatType
 {
 
-    FloatType(float num) : value( new float (num) ){}
+    private:
+        float* value;
+        FloatType& powInternal(float arg);
 
-    ~FloatType()
-    {
-        delete value;
-        value = nullptr;
-    }
+    public:    
+        FloatType(float num) : value( new float (num) ){}
 
-    operator float() { return *value;}
+        ~FloatType()
+        {
+            delete value;
+            value = nullptr;
+        }
 
-    FloatType& add( float rhs );
-    FloatType& subtract( float rhs );
-    FloatType& multiply( float rhs );
-    FloatType& divide( float rhs );
+        operator float() const { return *value;}
 
-    private:
-        float* value; 
+        FloatType& add( float rhs );
+        FloatType& subtract( float rhs );
+        FloatType& multiply( float rhs );
+        FloatType& divide( float rhs );
+
+        FloatType& pow(const IntType& rhs);
+
+        FloatType& pow(const FloatType& rhs);
+
+        FloatType& pow(const DoubleType& rhs);
+
+        FloatType& pow(float rhs);
+    
+ 
 };
 
 struct DoubleType
 {
-    DoubleType(double num) : value( new double (num) ){}
+    
+    private:
+        double* value;
+        DoubleType& powInternal(double arg);
 
-    ~DoubleType()
-    {
-        delete value;
-        value = nullptr;
-    }
+    public:
```

Comment:
if you move these private members to the end of the class, you can eliminate this `public` keyword, since struct is public by default
less typing = faster coding
- [ ] revise